### PR TITLE
Feat/testnet deployment

### PR DESCRIPTION
**Pull request by bitbeckers**
* added config and readme for goerli deployment and proxy validation
* Added contract sizer plugin to evaluate -well- contract sizes
* Added an upgrade with the name field to support ERC1155/ERC721 marketplaces
* Improved deployment logic for local testing vs. remote deployments

Partial feature of: https://github.com/protocol/hypercerts-protocol/issues/22
closes #27

### CODE REVIEW

File: contracts/HypercertMinterV1.sol [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+/// @title Hypercert Minting logic
+/// @notice Contains functions and events to initialize and issue a hypercert
+/// @author bitbeckers, mr_bluesky
+//TODO Merge in changes to main contract. This upgrade is used to aggregate changes
+contract HypercertMinterV1 is HypercertMinterV0 {
+    string public constant NAME = "Impact hypercertificates";
+
+    /// @notice gets the current version of the contract
+    function version() public pure virtual override returns (uint256) {
+        return 0;
```
Comment (moconnell) [inline contracts/HypercertMinterV1.sol:15]: 0 not 1? 🤔

File: deploy/001_HypercertMinter_Upgrade.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import { ethers, upgrades } from "hardhat";
+import { DeployFunction } from "hardhat-deploy/types";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+
+const deploy: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const { deployments } = hre; // we get the deployments and getNamedAccounts which are provided by hardhat-deploy.
+  const { save, get } = deployments; // The deployments field itself contains the deploy function.
+
+  try {
+    await get("HypercertMinterV0");
```
Comment (moconnell) [inline deploy/001_HypercertMinter_Upgrade.ts:10]: I would stick these well-known strings in a constants file personally

File: package.json [hunk 3]
```diff
@@ -51,15 +52,15 @@
     "mocha": "^10.0.0",
     "pinst": "^3.0.0",
     "prettier": "^2.7.1",
-    "prettier-plugin-solidity": "^1.0.0-beta.24",
+    "prettier-plugin-solidity": "^1.0.0-dev.23",
```
Comment (moconnell) [inline package.json:55]: Is this actually going backwards? 😁

File: deploy/001_HypercertMinter_Upgrade.ts (hunk 2 above)
Comment (bitbeckers) [reply deploy/001_HypercertMinter_Upgrade.ts:10]: will do

File: contracts/HypercertMinterV1.sol (hunk 1 above)
Comment (bitbeckers) [reply contracts/HypercertMinterV1.sol:15]: yeah it broke the test since the upgrade is also rolled out. Sloppy of me, I'll make it 1 again

**Pull request merged**